src/rosdiscover/observer/ros1.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+from typing import Callable, Collection, Dict, Mapping
+
+from loguru import logger
+from roswire import AppInstance
+
+from .observer import Observer
+from .ros1_actions import ActionCandidate, separate_topics_from_action
+from .. import Config
+from ..interpreter import NodeContext, ParameterServer, SystemSummary
+
+_NODES_TO_FILTER_OUT = {'/rosout'}
+_TOPICS_TO_FILTER_OUT = {'/rosout', '/rosout_agg'}
+_SERVICES_TO_FILTER_OUT = {'set_logger_level', 'get_loggers'}
+
+
+class ROS1Observer(Observer):
+
+    def __init__(self, app: AppInstance, config: Config):
+        super().__init__(app, config)
+
+    def observe_and_summarise(self) -> SystemSummary:
+        """
+        Takes a snapshot of the ROS system running in the container referred to by
+        :code:``self.app_instance`` and returns the architecture summary for that system.
+
+        Returns
+        -------
+        The summary of the instantaneous state of the ROS system
+
+        Raises
+        ------
+        TimeoutException
+            If a connection cannot be made to the container
+        """
+        try:
+            nodecontexts: Dict[str, NodeContext] = {}
+            with self._app_instance.ros1() as ros:
+                nodes = [n for n in ros.nodes if n not in _NODES_TO_FILTER_OUT]
+                info = ros.state
+                # Create a context for each node
```

Comment:
To improve the readability of the code, it might help to introduce a separate method that creates the context for a particular node (e.g., `_fetch_node_context(node: Node) -> NodeContext)`.